internal/test/consume_contract_test.go

```diff
@@ -1,7 +1,7 @@
 // Copyright The OpenTelemetry Authors
 // SPDX-License-Identifier: Apache-2.0
 
-package otlpexporter
+package test
```

Comment:
If we move this we create again the same circular dependency and even more:

otlpexporter -> main otel package (with internal)
otlpreceiver -> main otel package (with internal)

Now after this change `main otel package (with internal)` depends on `otlpexporter` and `otlpreceiver`

Reply:
Can we create a new module for those tests? I was asking @codeboten if there is a package we can use

Reply:
We can definitely have some "e2e" module that we don't "export" (in internal).

Reply:
Thanks! I'll get on that right away.

Reply:
ok this is done, the diff is much better